setup.py

```diff
@@ -1,39 +1,67 @@
-from setuptools import setup, find_packages
+"""setuptools config for nowplaypadgen"""
+
 import os
 import re
+import sys
+
+import nowplaypadgen
+
+from setuptools import setup, find_packages
+
 
 def readme():
-    with open('README.rst') as f:
-        return f.read()
+    """Load README.rst file for description"""
+    with open('README.rst') as readme_file:
+        return readme_file.read()
+
 
 def get_version():
-    VERSIONFILE = os.path.join('nowplaypadgen', '__init__.py')
-    initfile_lines = open(VERSIONFILE, 'rt').readlines()
-    VSRE = r"^__version__ = ['\"]([^'\"]*)['\"]"
+    """Get version from main module"""
+    version_file = os.path.join('nowplaypadgen', '__init__.py')
+    initfile_lines = open(version_file, 'rt').readlines()
     for line in initfile_lines:
-        mo = re.search(VSRE, line, re.M)
-        if mo:
-            return mo.group(1)
-    raise RuntimeError('Unable to find version string in %s.' % VERSIONFILE)
+        matches = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", line, re.M)
+        if matches:
+            return matches.group(1)
+    raise RuntimeError('Unable to find version string in %s.' % version_file)
+
+
+def pytest_runner():
+    """Require pytest-runner for invocations of setup.py that will invoke it."""
+    needs_pytest = {'pytest', 'test', 'ptr'}.intersection(sys.argv)
+    return ['pytest-runner~=2.11.1'] if needs_pytest else []
+
+INSTALL_REQUIRES = [
```

Comment:
Are this lists personal preferences or are they a commonly accepted pattern?

Reply:
AFAIK they are the django way to do this instead of creating elaborate lists directly in the setup() call.